### PR TITLE
Filter Products

### LINKED ISSUE #10: Add Filters to Products page
- Add Filters to Products page
- Should be able to filter products based on the filters
- Should be responsive

### PR DESCRIPTION
**Pull request by ttahm3d**
- FIltering Products 
  - Sort by
  - Filter out of stock
  - filter by category
  - filter by price range
  - filter by rating
  - filter by fast delivery

Preview
localhost_3000_products (4)

### CODE REVIEW

File: src/context/products/utils.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  products.filter((product) => product.price >= state.minPrice);
+
+const filterOutOutOfStockProducts = (state, products) =>
+  state.showOnlyInStock
+    ? products.filter((product) => product.count !== 0)
+    : products;
+
+const showFastDeliveryProducts = (state, products) =>
+  state.fastDelivery
+    ? products.filter((product) => product.aegisAssured === true)
```
Comment (sadathshariff) [inline src/context/products/utils.jsx]: Here the checking is not required  it will filter out if its true

**Review by sadathshariff — COMMENTED** [review]
The Code Looks Good like the naming convention you have used

File: src/context/products/Context.jsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  const [state, dispatch] = useReducer(productsReducer, initialState);
+
+  const [products, setProducts] = useState([]);
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const { data } = await axios.get("/api/products");
+        setProducts(data?.products);
+      } catch (e) {
+        console.log("error", e);
```
Comment (aishwarya-bm) [inline src/context/products/Context.jsx:43]: Maybe replace console log with an alert message.

**Review by aishwarya-bm — APPROVED** [review]
Good naming convention and reuse of components. Clean code.

File: src/context/products/Context.jsx (hunk 2 above)
Comment (ttahm3d) [reply src/context/products/Context.jsx:43]: this is temporary for now will refactor later

**Pull request merged**